metadata/metadata.go

```diff
@@ -310,18 +320,101 @@ func isDirectDependency(ctx context.Context, lookup ResourceLookup, dependency,
 	return deps.Has(ctx, depId)
 }
 
+type DeletionMode int
+
+const (
+	ExcludeDeleted DeletionMode = iota // default: exclude deleted
+	IncludeDeleted
+	DeletedOnly
+)
+
+type ResourceLookupType string
+
+type ResourceLookupOption interface {
+	Type() ResourceLookupType
+}
+
+const DeleteLookupOptionType ResourceLookupType = "Deletion"
+
+type DeleteLookupOption struct {
+	deletionMode DeletionMode
+}
+
+func (opt DeleteLookupOption) Type() ResourceLookupType {
+	return DeleteLookupOptionType
+}
+
+type ResourceLookupOptions struct {
```

Comment:
This is super confusing naming given the above, maybe call this parsedResourceLookupConfig or something to make it clear its for internal use